pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -158,19 +131,19 @@ func (pc *PodCache) onEvent(_, pod *v1.Pod, ev model.Event) error {
 	key := config.NamespacedName(pod)
 	switch ev {
 	case model.EventAdd:
-		if shouldPodBeInEndpoints(pod) && IsPodReady(pod) {
+		if endpointsliceutil.ShouldPodBeInEndpoints(pod, false) && IsPodReady(pod) {
```

Comment:
https://github.com/istio/istio/pull/42156 updated ShouldPodBeInEndpoints with latest k8s but the function signature is inconsistent with the upstream code https://github.com/kubernetes/kubernetes/blob/133be895e0de2994353703e082f92b6b946ac0e2/pkg/controller/util/endpoint/controller_utils.go#L141. Is it intentional or a bug? @hzxuzhonghu @ramaraochavali

The unit tests always pass when I switch `false` to `true`.

https://github.com/kubernetes/kubernetes/pull/90783 clarified behavior of publishNotReadyAddresses like the following:

```
(base) ➜  ~ kubectl explain svc.spec.publishNotReadyAddresses
KIND:       Service
VERSION:    v1

FIELD: publishNotReadyAddresses <boolean>


DESCRIPTION:
    publishNotReadyAddresses indicates that any agent which deals with endpoints
    for this Service should disregard any indications of ready/not-ready. The
    primary use case for setting this field is for a StatefulSet's Headless
    Service to propagate SRV DNS records for its Pods for the purpose of peer
    discovery. The Kubernetes controllers that generate Endpoints and
    EndpointSlice resources for Services interpret this to mean that all
    endpoints are considered "ready" even if the Pods themselves are not. Agents
    which consume only Kubernetes generated endpoints through the Endpoints or
    EndpointSlice resources can safely assume this behavior.
```


FYI: 
- https://github.com/kubernetes/kubernetes/blob/60c4c2b2521fb454ce69dee737e3eb91a25e0535/staging/src/k8s.io/endpointslice/reconciler.go#L171
- https://github.com/kubernetes/kubernetes/blob/60c4c2b2521fb454ce69dee737e3eb91a25e0535/pkg/controller/endpoint/endpoints_controller.go#L425

Reply:
what do you mean by signature?

Reply:
k/k: `func ShouldPodBeInEndpoints(pod *v1.Pod, includeTerminating bool) bool`
istio: `func shouldPodBeInEndpoints(pod *v1.Pod) bool `

Reply:
It is intended, always ignore terminating pod

Reply:
What will happen if we don't ignore it? the upstream https://github.com/kubernetes/kubernetes/blob/60c4c2b2521fb454ce69dee737e3eb91a25e0535/staging/src/k8s.io/endpointslice/reconciler.go#L171 are always considered "ready" even if the Pods themselves are not.

I'm not familiar with the Istio project, can you help me to clarify the above questions? Thanks!

Reply:
This means when a pod is draining, we donot generate endpoint with it.

Reply:
I see, thank you.